test/it/EditVatLinkAcceptanceSpec.scala

```diff
@@ -74,4 +104,110 @@ class EditVatLinkAcceptanceSpec extends ServerAcceptanceSpec with WireMockSbrCon
+      And(s"And the VAT record is also being edited by somebody else, resulting in a conflict")
+      stubSbrControlApiFor(aVatParentLinkEditRequest(withVatRef = TargetVAT, withPeriod = TargetPeriod)
+        .withHeader(HeaderNames.CONTENT_TYPE, equalTo(PatchJson))
+        .withRequestBody(equalToJson(VATEditParentLinkPatchBody))
+        .willReturn(aConflictResponse()))
+
+      When(s"an edit request for the VAT unit with $TargetVAT is requested for $TargetPeriod")
+      val response = await(wsClient
+        .url(s"/v1/periods/${Period.asString(TargetPeriod)}/edit/vats/${TargetVAT.value}")
+        .withHeaders((HeaderNames.CONTENT_TYPE, JSON))
+        .post(VATEditParentLinkPostBody))
+
+      Then(s"the status of the edit of the parent LEU from $ParentLEU to $NewParentLEU will be sent back")
+      response.status shouldBe CONFLICT
+    }
+  }
+
+  feature("edit an existing VAT records parent link where general errors are occurring on sbr-control-api") {
+    scenario("by VAT reference (vatref) and UBRN for a specific period") { wsClient =>
+      Given(s"a VAT record with $TargetVAT for $TargetPeriod does not exist")
+      stubSbrControlApiFor(aVatParentLinkEditRequest(withVatRef = TargetVAT, withPeriod = TargetPeriod)
+        .withHeader(HeaderNames.CONTENT_TYPE, equalTo(PatchJson))
+        .withRequestBody(equalToJson(VATEditParentLinkPatchBody))
+        .willReturn(anInternalServerError()))
+
+      When(s"an edit request for the VAT unit with $TargetVAT is requested for $TargetPeriod")
+      val response = await(wsClient
+        .url(s"/v1/periods/${Period.asString(TargetPeriod)}/edit/vats/${TargetVAT.value}")
+        .withHeaders((HeaderNames.CONTENT_TYPE, JSON))
+        .post(VATEditParentLinkPostBody))
+
+      Then(s"the status of the edit of the parent LEU from $ParentLEU to $NewParentLEU will be sent back")
+      response.status shouldBe INTERNAL_SERVER_ERROR
+    }
+  }
+
+  feature("edit an existing VAT records parent link using invalid JSON") {
+    scenario("by VAT reference (vatref) and UBRN for a specific period") { wsClient =>
+      When(s"an edit request for the VAT unit with $TargetVAT is requested for $TargetPeriod")
+      And(s"And the edit JSON is invalid (extra closing brace)")
+      val response = await(wsClient
+        .url(s"/v1/periods/${Period.asString(TargetPeriod)}/edit/vats/${TargetVAT.value}")
+        .withHeaders((HeaderNames.CONTENT_TYPE, JSON))
+        .post(InvalidVATEditParentLinkPostBody))
+
+      Then(s"the status of the edit of the parent LEU from $ParentLEU to $NewParentLEU will be sent back")
+      response.status shouldBe BAD_REQUEST
+    }
+  }
+
+  feature("edit an existing VAT records parent link using JSON that cannot be processed by sbr-control-api") {
+    scenario("by VAT reference (vatref) and UBRN for a specific period") { wsClient =>
+      When(s"an edit request for the VAT unit with $TargetVAT is requested for $TargetPeriod")
```

Comment:
Probably better style as a Given When

Given a valid edit request must contain a from UBRN and a to UBRN
When a request is made that contains two from UBRNs and no to UBRN
Then ...